src/App.tsx

```diff
@@ -15,48 +17,114 @@ export const goodsFromServer = [
   'Garlic',
 ];
 
-export const App: React.FC = () => {
+enum SortType {
+  NONE,
+  ALPHABET,
+  LENGTH,
+}
+
+type ReorderOptions = {
+  sortType: SortType,
+  isReversed: boolean,
+};
+
+export const App: FC = () => {
+  const getReorderedGoods = (
+    goods: string[],
+    { sortType, isReversed }: ReorderOptions,
+  ) => {
+    const visibleGoods = [...goods];
+
+    visibleGoods.sort((firstItem, secondItem) => {
+      switch (sortType) {
+        case SortType.ALPHABET:
+          return firstItem.localeCompare(secondItem);
+
+        case SortType.LENGTH:
+          return firstItem.length - secondItem.length;
+
+        default:
+          return 0;
+      }
+    });
+
+    if (isReversed) {
+      visibleGoods.reverse();
+    }
+
+    return visibleGoods;
+  };
+
+  const [sortType, setSortType] = useState(SortType.NONE);
+  const [isReversed, setReverse] = useState(false);
+  const shouldShowReset = isReversed || sortType !== SortType.NONE;
+
+  const sortBy = (sortByKey: SortType) => {
+    setSortType(sortByKey);
+  };
```

Comment:
This function does nothing. You can use `setSortType(SortType.ALPHABET)` without this function